src/Compilers/CSharp/Portable/Symbols/Symbol_Attributes.cs

```diff
@@ -418,6 +422,97 @@ internal virtual void PostDecodeWellKnownAttributes(ImmutableArray<CSharpAttribu
             Debug.Assert(lazyCustomAttributesBag.IsSealed);
             diagnostics.Free();
             return lazyAttributesStoredOnThisThread;
+
+            void removeObsoleteDiagnosticsForForwardeTypes(ImmutableArray<CSharpAttributeData> boundAttributes, ImmutableArray<AttributeSyntax> attributesToBind, ref BindingDiagnosticBag diagnostics)
```

Comment:
Worth asserting that `attributesToBind` and `boundAttributes` are the same length?

Reply:
I think it is "obvious" from the code above, but I will add one since I need to fix a typo in this code anyway.